test/app-luatest/fiber_test.lua

```diff
@@ -57,3 +57,26 @@ g.test_gh_9406_shutdown_with_lingering_fiber_join = function()
     local cmd = string.format('%s -e "%s"', tarantool_bin, script)
     t.assert(os.execute(cmd) == 0)
 end
+
+g.test_gh_10187_no_memory_leak_on_dead_fiber_search = function()
```

Comment:
I think that this fix does deserve a changelog entry.

Reply:
Done.

```md
## bugfix/box

* Fixed memory leaks on using dead fiber (gh-10187).
```